operator/validator/controller.go

```diff
@@ -508,15 +511,22 @@ func (c *controller) StartValidators() {
 	}
 }
 
-func (c *controller) selfSubnets() records.Subnets {
+// selfSubnets calculates the operator's subnets by adding up the fixed subnets and the active committees
+// it recvs big int to avoid unnecessary conversions, if intCid is nil it will allocate a new big int
+func (c *controller) selfSubnets(intCid *big.Int) records.Subnets {
```

Comment:
in this context its just a buffer and has no usage outside this func, so to avoid confusion i'd call it just "buffer" and clarify it's for re-using memory (don't have to go into much details)

Reply:
taking inspiration from https://pkg.go.dev/bytes#Buffer

which has a similar pattern to what you did where if you dont pass `buf` it will allocate for you